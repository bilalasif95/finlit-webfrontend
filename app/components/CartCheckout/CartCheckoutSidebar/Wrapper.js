import styled from 'styled-components';

const Wrapper = styled.div`
  border: 1px solid #e4e4e4;
  border-radius: 0.313rem;
  padding: 2rem;
  margin: 0 0 2rem;
  width: 100%;
  .heading {
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin: 0 0 1rem;
    h4 {
      color: #484848;
      font-size: 1.25rem;
      margin: 0;
      position: relative;
    }
    svg {
      color: #484848;
      font-size: 1.5rem;
    }
  }
  p {
    color: #8b8b8b;
    font-size: 0.875rem;
    margin: 0 0 1rem;
  }
  fieldset {
    margin: 0 0 1rem;
    padding: 0;
    .form-check {
      align-items: center;
      cursor: pointer;
      display: flex;
      justify-content: space-between;
      margin: 0 0 0.5rem;
      label {
        color: #484848;
        font-size: 0.875rem;
        margin: 0;
      }
    }
    .form-check-input:focus {
      box-shadow: none;
    }
    .form-check-input:checked {
      background-color: transparent;
      border-color: #255f2b;
    }
  }
  .form-group {
    margin: 0 0 1rem;
    label {
      color: #484848;
      font-size: 0.875rem;
    }
    .form-control {
      border-top-right-radius: 0.25rem !important;
      border-bottom-right-radius: 0.25rem !important;
      &:focus {
        box-shadow: none;
      }
    }
  }
  .form-check {
    label {
      color: #8b8b8b;
    }
  }
  ul {
    margin: 1rem 0 0;
    li {
      align-items: center;
      display: flex;
      justify-content: space-between;
      padding: 0 0 1rem;
      p {
        color: #484848;
        font-size: 0.875rem;
        line-height: 16px;
        margin: 0;
        width: 50%;
        &:last-child {
          text-align: right;
        }
      }
      h4 {
        color: #ec5252;
        font-size: 1.25rem;
        font-weight: 600;
        line-height: 24px;
        margin: 0;
        text-align: right;
        width: 50%;
      }
    }
  }
  .term_service {
    p {
      color: #8b8b8b;
      font-size: 0.625rem;
      line-height: 12px;
    }
    a {
      color: #46a135;
      font-size: 0.625rem;
      line-height: 12px;
      text-decoration: none;
    }
  }
  button {
    background-color: #46a135;
    border: 1px solid #46a135;
    border-radius: 0.313rem;
    color: #fff;
    font-size: 0.875rem;
    font-weight: 550;
    line-height: 17px;
    margin: 1rem 0 0;
    padding: 0.85rem;
    width: 100%;
    &:hover {
      background-color: #46a135;
      border: 1px solid #46a135;
    }
    &:focus {
      background-color: #46a135;
      border: 1px solid #46a135;
      box-shadow: none;
    }
  }
  .secure_payment {
    align-items: center;
    display: flex;
    justify-content: center;
    margin: 1rem 0 0;
    width: 100%;
    svg {
      color: #8b8b8b;
      width: 1.25rem;
      height: 1.25rem;
    }
    p {
      margin: 0;
    }
  }
  @media (max-width: 960px) and (min-width: 768px) {
    margin: 0 0 1.5rem;
    padding: 1.25rem;
  }
  @media (max-width: 767px) {
    margin: 1rem 0;
    padding: 1.25rem;
  }
  @media (max-width: 600px) {
    padding: 1rem;
  }
`;

export default Wrapper;
