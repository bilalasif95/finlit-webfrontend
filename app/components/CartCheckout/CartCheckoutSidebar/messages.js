/*
 * Cart Checkout Sidebar Messages
 *
 * This contains all the text for the CartCheckoutSidebar component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.components.CartCheckoutSidebar';

export default defineMessages({
  PaymentDetails: {
    id: `${scope}.PaymentDetails`,
    defaultMessage: `Payment Details`,
  },
  CompletePurchase: {
    id: `${scope}.CompletePurchase`,
    defaultMessage: `Complete your purchase by providing your payment details.`,
  },
  EmailAddress: {
    id: `${scope}.EmailAddress`,
    defaultMessage: `Email Address`,
  },
  CreditCard: {
    id: `${scope}.CreditCard`,
    defaultMessage: `Credit Card / Visa Card`,
  },
  PayPal: {
    id: `${scope}.PayPal`,
    defaultMessage: `PayPal`,
  },
  CardholderName: {
    id: `${scope}.CardholderName`,
    defaultMessage: `Cardholder name`,
  },
  CardNumber: {
    id: `${scope}.CardNumber`,
    defaultMessage: `Card Number`,
  },
  MMYYCVC: {
    id: `${scope}.MMYYCVC`,
    defaultMessage: `MM / YY CVC`,
  },
  RememberCard: {
    id: `${scope}.RememberCard`,
    defaultMessage: `Remember this card`,
  },
  Summary: {
    id: `${scope}.Summary`,
    defaultMessage: `Summary`,
  },
  OriginalPrice: {
    id: `${scope}.OriginalPrice`,
    defaultMessage: `Original Price:`,
  },
  CouponDiscounts: {
    id: `${scope}.CouponDiscounts`,
    defaultMessage: `Coupon Discounts:`,
  },
  Total: {
    id: `${scope}.Total`,
    defaultMessage: `Total:`,
  },
  YouAgree: {
    id: `${scope}.YouAgree`,
    defaultMessage: `By completing your purchase you agree to these`,
  },
  TermsService: {
    id: `${scope}.TermsService`,
    defaultMessage: `Terms of Service.`,
  },
  Pay: {
    id: `${scope}.Pay`,
    defaultMessage: `Pay`,
  },
  PaymentsEncrypted: {
    id: `${scope}.PaymentsEncrypted`,
    defaultMessage: `Payments are secure and encrypted`,
  },
});
