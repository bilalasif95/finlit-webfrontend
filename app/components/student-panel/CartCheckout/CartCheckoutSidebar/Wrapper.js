import styled from 'styled-components';

const Wrapper = styled.div`
  .error {
    color: red;
  }
  margin-top: 31px;
  .heading {
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin: 0 0 1rem;
    img {
      margin-right: 14px;
    }
    h4 {
      color: #404041;
      font-size: 1.25rem;
      margin: 0;
      position: relative;
    }
    svg {
      color: #8b8b8b;
      font-size: 1.5rem;
    }
  }
  p {
    color: #8b8b8b;
    font-size: 0.875rem;
    margin: 0 0 1rem;
  }
  .cardSelect {
    padding: 1.5rem 2rem;
    .form-check {
      align-items: center;
      cursor: pointer;
      display: flex;
      justify-content: space-between;
      margin: 0;
      label {
        color: #484848;
        font-size: 0.875rem;
        margin: 0;
      }
    }
    .form-check-input:focus {
      box-shadow: none;
    }
    .form-check-input:checked {
      background-color: transparent;
      border-color: #255f2b;
    }
    .form-check-input:checked[type='radio'] {
      background-image: url(https://cdn.discordapp.com/attachments/548125906472796170/887240829683507240/checkdot.png);
      background-size: 6px;
      background-position: center;
    }
  }
  .card_payment {
    margin: 1rem 0 0;
  }
  .paypal {
    border: 1px solid #276F86;
    border-radius: 5px;
    background-color: #F8FFFF;
    padding: 1rem;
    margin: 1.5rem 0 0; 
    p {
      margin-bottom: 0px;
      position: relative;
      padding-left: 30px;
      color: #276F86;
      font-weight: 700;
      font-size: 14px;
      img {
        position: absolute;
        left: 0;
        top: 3px;
      }
    }
  }
  // .paypal {
  //   background-color: #fbfbfb;
  //   padding: 1rem;
  //   p {
  //     color: #484848;
  //     font-size: 0.875rem;
  //     margin: 0;
  //   }
  // }
  .form-group {
    margin: 0 0 1rem;
    label {
      color: #484848;
      font-size: 0.875rem;
    }
    .form-control {
      border-top-right-radius: 0.25rem !important;
      border-bottom-right-radius: 0.25rem !important;
      &:focus {
        box-shadow: none;
      }
      &::placeholder {
        color: #e6e6e6;
      }
    }
  }
  .form-check {
    label {
      color: #8b8b8b;
    }
    .form-check-input:checked {
      background-color: transparent;
      border-color: #47a135;
    }
    .form-check-input:focus {
      box-shadow: none;
    }
  }
  ul {
    margin: 1rem 0 0;
    padding: 0;
    li {
      align-items: flex-start;
      display: flex;
      justify-content: space-between;
      list-style: none;
      padding: 0 0 1rem;
      p {
        color: #484848;
        font-size: 0.875rem;
        line-height: 16px;
        margin: 0;
        width: 50%;
        &:last-child {
          text-align: right;
        }
      }
      h4 {
        color: #ec5252;
        font-size: 1.25rem;
        font-weight: 600;
        line-height: 24px;
        margin: 0;
        text-align: right;
        width: 50%;
      }
    }
  }
  .term_service {
    p {
      color: #8b8b8b;
      font-size: 0.625rem;
      line-height: 12px;
      margin-bottom: 40px;
    }
    a {
      color: #46a135;
      font-size: 0.625rem;
      line-height: 12px;
      text-decoration: none;
    }
  }
  button {
    background-color: #46a135;
    border: 1px solid #46a135;
    border-radius: 0.313rem;
    color: #fff;
    font-size: 0.875rem;
    font-weight: 550;
    line-height: 17px;
    margin: 0 0 1.25rem;
    padding: 1rem 0.85rem;
    width: 100%;
    &:hover {
      background-color: #46a135;
      border: 1px solid #46a135;
    }
    &:focus {
      background-color: #46a135;
      border: 1px solid #46a135;
      box-shadow: none;
    }
  }
  .secure_payment {
    align-items: center;
    display: flex;
    margin: 1rem 0 0;
    width: 100%;
    svg {
      color: #8b8b8b;
      width: 1.25rem;
      height: 1.25rem;
    }
    p {
      margin: 0;
    }
  }
  @media (max-width: 960px) and (min-width: 768px) {
    margin: 0 0 1.5rem;
    padding: 1.25rem;
  }
  @media (max-width: 767px) {
    margin: 1rem 0;
    padding: 1.25rem;
  }
  @media (max-width: 600px) {
    padding: 0;
  }
  @media (max-width: 768px) {
    .payment-box {
      .heading {
        h4 {
          font-size: 16px;
          img {
            margin-right: 6px;
          }
        }
      }
      .innerBox {
       .paypalBox {
          padding: 10px 10px;
          margin-bottom: 20px;
        }
      }
      .innerBox, .cardSelect {
        padding: 1rem;
      }
      .summary {
        h4 {
          font-size 16px;
        }
        .list {
          li {
            padding: 15px 0px;
            p {
              span {
                width: 162px;
                font-size: 12px;
              }
            } 
          } 
        } 
      } 
      .secure_payment {
        p {
          font-size: 12px;
        }
      }
      .term_service {
        p {
          font-size: 12px;
          margin-bottom: 20px;
        }
      }
    }
  }
  
`;

export default Wrapper;
