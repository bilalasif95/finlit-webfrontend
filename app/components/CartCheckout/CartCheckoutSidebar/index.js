/*
 * Cart Checkout Sidebar Component
 */
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Row, Col, FormGroup, Input, Label, Button } from 'reactstrap';
import { IoMdLock } from 'react-icons/io';
import { BsArrowLeftShort } from 'react-icons/bs';
import { Link } from 'react-router-dom';
import Wrapper from './Wrapper';
import messages from './messages';
import mastercard from '../../../images/mastercard.svg';
import paypal from '../../../images/paypal.svg';

function CartCheckoutSidebar() {
  return (
    <Row>
      <Col lg={12}>
        <Wrapper>
          <div className="heading">
            <h4>
              <FormattedMessage {...messages.PaymentDetails} />
            </h4>
            <Link to="/">
              <BsArrowLeftShort />
            </Link>
          </div>
          <p>
            <FormattedMessage {...messages.CompletePurchase} />
          </p>
          <FormGroup>
            <Label for="email">
              <FormattedMessage {...messages.EmailAddress} />
            </Label>
            <Input
              type="email"
              name="email"
              id="email"
              placeholder="Enter email"
            />
          </FormGroup>
          <FormGroup tag="fieldset">
            <FormGroup check>
              <Label check>
                <Input type="radio" name="radio1" />
                <FormattedMessage {...messages.CreditCard} />
              </Label>
              <div className="icon">
                <img src={mastercard} alt="Master card" />
              </div>
            </FormGroup>
            <FormGroup check>
              <Label check>
                <Input type="radio" name="radio1" />
                <FormattedMessage {...messages.PayPal} />
              </Label>
              <div className="icon">
                <img src={paypal} alt="Paypal" />
              </div>
            </FormGroup>
          </FormGroup>
          <FormGroup>
            <Label for="cardholdername">
              <FormattedMessage {...messages.CardholderName} />
            </Label>
            <Input
              type="text"
              name="cardholdername"
              id="cardholdername"
              placeholder="Enter name"
            />
          </FormGroup>
          <FormGroup>
            <Label for="cardnumber">
              <FormattedMessage {...messages.CardNumber} />
            </Label>
            <Input
              type="text"
              name="cardnumber"
              id="cardnumber"
              placeholder="Enter number"
            />
          </FormGroup>
          <FormGroup>
            <Label for="cvc">
              <FormattedMessage {...messages.MMYYCVC} />
            </Label>
            <Input
              type="text"
              name="cvc"
              id="cvc"
              placeholder="mm - yy - CVC"
            />
          </FormGroup>
          <FormGroup check>
            <Label check className="check_label">
              <Input type="checkbox" />
              <FormattedMessage {...messages.RememberCard} />
            </Label>
          </FormGroup>
          <div className="summary">
            <h4>
              <FormattedMessage {...messages.Summary} />
            </h4>
            <ul clasName="list">
              <li>
                <p>
                  <FormattedMessage {...messages.OriginalPrice} />
                </p>
                <p>$94.99</p>
              </li>
              <li>
                <p>
                  <FormattedMessage {...messages.CouponDiscounts} />
                </p>
                <p>-$74.00</p>
              </li>
              <li>
                <p>
                  <FormattedMessage {...messages.Total} />
                </p>
                <h4>$20.99</h4>
              </li>
            </ul>
          </div>
          <div className="term_service">
            <p>
              <FormattedMessage {...messages.YouAgree} />
              &nbsp;
              <Link to="/">
                <FormattedMessage {...messages.TermsService} />
              </Link>
            </p>
          </div>
          <Button>
            <FormattedMessage {...messages.Pay} />
            &nbsp;$20.00
          </Button>
          <div className="secure_payment">
            <IoMdLock />
            <p>
              <FormattedMessage {...messages.PaymentsEncrypted} />
            </p>
          </div>
        </Wrapper>
      </Col>
    </Row>
  );
}

export default CartCheckoutSidebar;
