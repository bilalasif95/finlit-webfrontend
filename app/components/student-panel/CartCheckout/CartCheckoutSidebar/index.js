/*
 * Cart Checkout Sidebar Component
 */
import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';
import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';
import { Row, Col, FormGroup, Input, Label, Button } from 'reactstrap';
import { IoMdLock } from 'react-icons/io';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import history from 'utils/history';
import mastercard from '../../../../images/mastercard.svg';
import paypalicon from '../../../../images/paypal.svg';
import { axiosHeader } from '../../../../utils/axiosHeader';
import { API } from '../../../../config/config';
import Wrapper from './Wrapper';
import messages from './messages';
function CartCheckoutSidebar(props) {
  const [paymentType, setPaymentType] = useState({ card: true, paypal: false });
  const [loader, setLoader] = useState(false);
  const [checkoutInfo, setCheckoutInfo] = useState({
    email: '',
    cardHolderName: '',
    cardNumber: '',
    MMYY: '',
    CVC: '',
    rememberCard: false,
  });
  const [errors, setErrors] = useState({});

  const handleChange = type => {
    if (type === 'card') {
      setPaymentType({ card: true, paypal: false });
    } else {
      setPaymentType({ card: false, paypal: true });
    }
  };

  const handleChangeEvent = event => {
    if (event.target.type === 'checkbox') {
      setCheckoutInfo({
        ...checkoutInfo,
        [event.target.name]: event.target.checked,
      });
    } else if (event.target.type === 'file') {
      setCheckoutInfo({
        ...checkoutInfo,
        [event.target.name]: event.target.files[0],
      });
    } else {
      setCheckoutInfo({
        ...checkoutInfo,
        [event.target.name]: event.target.value,
      });
    }
  };

  const handleCheckout = () => {
    if (Object.keys(validator(checkoutInfo)).length > 0) {
      setErrors(validator(checkoutInfo));
      setTimeout(() => {
        setErrors({});
      }, 4000);
    } else {
      setLoader(true);
      axios
        .get(`${API}api/cart/checkout`, axiosHeader)
        .then(() => {
          toast.success(`Checkout successful. You will get an email shortly`);
          setLoader(false);
          setTimeout(() => {
            history.push('/');
          }, 5000);
        })
        .catch(err => {
          setLoader(false);
          setErrors(err.response && err.response.data.message);
        });
    }
  };

  const validator = values => {
    const error = {};
    if (!values.email) {
      error.email = 'Email is required';
    } else {
      const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
      const validEmail = regex.test(String(values.email).toLowerCase());
      if (!validEmail) {
        error.email = 'Invalid email';
      } else if (!values.cardHolderName) {
        error.cardHolderName = 'Card name is required';
      } else if (!values.cardNumber) {
        error.cardNumber = 'Card number is required';
      } else {
        const validateCard = values.cardNumber.toString().search('_');
        if (validateCard >= 0) {
          error.cardNumber = 'Invalid card number format';
        } else if (!values.MMYY) {
          error.MMYY = 'MM / YY  is required';
        } else {
          const validateMM = values.MMYY.toString().search('M');
          if (validateMM >= 0) {
            error.MMYY = 'Invalid format mm/yy';
          } else {
            const validateyy = values.MMYY.toString().search('Y');
            if (validateyy >= 0) {
              error.MMYY = 'Invalid format mm/yy';
            } else if (!values.CVC) {
              error.CVC = 'CVC is required';
            }
            // else if (isNaN(values.CVC)) {
            //   error.CVC = 'Invalid CVC format';
            // }
          }
        }
      }
    }
    return error;
  };
  return (
    <Row>
      <Col lg={12}>
        <Wrapper>
          <div className="payment-box">
            <div className="heading">
              <h4>
                <FormattedMessage {...messages.PaymentDetails} />
              </h4>
            </div>
            <p>
              <FormattedMessage {...messages.CompletePurchase} />
            </p>
            <FormGroup>
              <Label for="email">
                <FormattedMessage {...messages.EmailAddress} />
              </Label>
              <Input
                type="email"
                name="email"
                id="email"
                value={checkoutInfo.email}
                onChange={e => handleChangeEvent(e)}
                placeholder="Enter email"
                required
              />
              <Label for="email">
                {errors.email ? <p className="error"> {errors.email} </p> : ''}
              </Label>
            </FormGroup>
            <FormGroup tag="fieldset">
              <FormGroup check>
                <Label check>
                  <Input
                    type="radio"
                    name="paymentType"
                    onChange={() => handleChange('card')}
                    checked={paymentType.card}
                  />
                  <FormattedMessage {...messages.CreditCard} />
                </Label>
                <div className="icon">
                  <img
                    src={mastercard}
                    alt="Master card"
                    height="100%"
                    width="100%"
                  />
                </div>
              </FormGroup>
              <FormGroup check disabled>
                <Label check>
                  <Input
                    type="radio"
                    name="paymentType"
                    onChange={() => handleChange('Paypal')}
                    checked={paymentType.paypal}
                    disabled
                  />
                  <FormattedMessage {...messages.PayPal} />
                </Label>
                <div className="icon">
                  <img
                    src={paypalicon}
                    alt="Paypal"
                    height="100%"
                    width="100%"
                  />
                </div>
              </FormGroup>
            </FormGroup>
            {/* <div className="paypal">
              <p>
                <FormattedMessage {...messages.PayPalInfo} />
              </p>
            </div> */}
            <FormGroup>
              <Label for="cardholdername">
                <FormattedMessage {...messages.CardholderName} />
              </Label>
              <Input
                type="text"
                name="cardHolderName"
                id="cardholdername"
                value={checkoutInfo.cardHolderName}
                onChange={e => handleChangeEvent(e)}
                placeholder="Enter name"
              />
              <Label for="cardholdername">
                {errors.cardHolderName ? (
                  <p className="error"> {errors.cardHolderName} </p>
                ) : (
                  ''
                )}
              </Label>
            </FormGroup>
            <FormGroup>
              <Label for="cardnumber">
                <FormattedMessage {...messages.CardNumber} />
              </Label>
              <NumberFormat
                format="#### #### #### ####"
                mask="_"
                name="cardNumber"
                value={checkoutInfo.cardNumber}
                onChange={e => handleChangeEvent(e)}
                placeholder="Enter number"
                className="form-control"
              />
              <Label for="cardnumber">
                {errors.cardNumber ? (
                  <p className="error"> {errors.cardNumber} </p>
                ) : (
                  ''
                )}
              </Label>
            </FormGroup>
            <FormGroup>
              <Label for="mmyy">
                <FormattedMessage {...messages.MMYY} />
              </Label>
              <NumberFormat
                format="##/##"
                mask={['M', 'M', 'Y', 'Y']}
                id="mmyy"
                name="MMYY"
                value={checkoutInfo.MMYY}
                onChange={e => handleChangeEvent(e)}
                placeholder="mm - yy "
                className="form-control"
              />
              <Label for="mmyy">
                {errors.MMYY ? <p className="error"> {errors.MMYY} </p> : ''}
              </Label>
            </FormGroup>

            <FormGroup>
              <Label for="cvc">
                <FormattedMessage {...messages.CVC} />
              </Label>
              <NumberFormat
                format="###"
                placeholder="Enter CVC"
                name="CVC"
                value={checkoutInfo.CVC}
                onChange={e => handleChangeEvent(e)}
                className="form-control"
              />
              <Label for="cvc">
                {errors.CVC ? <p className="error">{errors.CVC}</p> : ''}
              </Label>
            </FormGroup>
            <FormGroup check>
              <Label check className="check_label">
                <Input
                  type="checkbox"
                  name="rememberCard"
                  checked={checkoutInfo.rememberCard}
                  onChange={e => handleChangeEvent(e)}
                />
                <FormattedMessage {...messages.RememberCard} />
              </Label>
            </FormGroup>
            <div className="summary">
              <h4>
                <FormattedMessage {...messages.Summary} />
              </h4>
              <ul clasName="list">
                <li>
                  <p>
                    <FormattedMessage {...messages.OriginalPrice} />
                  </p>
                  <p>${props.details.total}</p>
                </li>
                <li>
                  <p>
                    <FormattedMessage {...messages.CouponDiscounts} />
                  </p>
                  <p>-$000</p>
                </li>
                <li>
                  <p>
                    <FormattedMessage {...messages.Total} />
                  </p>
                  <h4>${props.details.total}</h4>
                </li>
              </ul>
            </div>
            <div className="term_service">
              <p>
                <FormattedMessage {...messages.YouAgree} />
                &nbsp;
                <Link to="/">
                  <FormattedMessage {...messages.TermsService} />
                </Link>
              </p>
            </div>
            <Button onClick={() => handleCheckout()}>
              {loader ? `loading...` : `Pay$${props.details.total}`}
            </Button>
            <div className="secure_payment">
              <IoMdLock />
              <p>
                <FormattedMessage {...messages.PaymentsEncrypted} />
              </p>
            </div>
          </div>
        </Wrapper>
      </Col>
      <ToastContainer />
    </Row>
  );
}
CartCheckoutSidebar.propTypes = {
  props: PropTypes.any,
  details: PropTypes.any,
};

export default CartCheckoutSidebar;
