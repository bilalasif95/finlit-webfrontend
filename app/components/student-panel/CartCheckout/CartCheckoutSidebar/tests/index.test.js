import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
// import { browserHistory, BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import history from 'utils/history';
import CartCheckoutSidebar from '../index';
// import configureStore from '../../../../../configureStore';
import Store2 from '../../../../../containers/reduxSetup/store';
describe('<CartCheckoutSidebar />', () => {
  let store;

  beforeAll(() => {
    // store = configureStore({}, browserHistory);
    store = Store2;
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            {/* <BrowserRouter> */}
            <Router history={history}>
              <CartCheckoutSidebar details={{ total: 30 }} />
            </Router>
            {/* </BrowserRouter> */}
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
