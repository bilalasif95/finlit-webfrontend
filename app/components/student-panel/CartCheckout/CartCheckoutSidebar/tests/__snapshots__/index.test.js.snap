// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCheckoutSidebar /> should render and match the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12"
  >
    <div
      className="Wrapper-sc-8on6gx-0 dvkBbM"
    >
      <div
        className="heading"
      >
        <h4>
          <span>
            Payment Details
          </span>
        </h4>
        <a
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M7.854 4.646a.5.5 0 010 .708L5.207 8l2.647 2.646a.5.5 0 01-.708.708l-3-3a.5.5 0 010-.708l3-3a.5.5 0 01.708 0z"
              fillRule="evenodd"
            />
            <path
              clipRule="evenodd"
              d="M4.5 8a.5.5 0 01.5-.5h6.5a.5.5 0 010 1H5a.5.5 0 01-.5-.5z"
              fillRule="evenodd"
            />
          </svg>
        </a>
      </div>
      <p>
        <span>
          Complete your purchase by providing your payment details.
        </span>
      </p>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="email"
        >
          <span>
            Email Address
          </span>
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          placeholder="Enter email"
          type="email"
        />
      </div>
      <fieldset
        className="form-group"
      >
        <div
          className="form-check"
        >
          <label
            className="form-check-label"
          >
            <input
              className="form-check-input"
              name="radio1"
              type="radio"
            />
            <span>
              Credit Card / Visa Card
            </span>
          </label>
          <div
            className="icon"
          >
            <img
              alt="Master card"
              src="IMAGE_MOCK"
            />
          </div>
        </div>
        <div
          className="form-check"
        >
          <label
            className="form-check-label"
          >
            <input
              className="form-check-input"
              name="radio1"
              onChange={[Function]}
              type="radio"
            />
            <span>
              PayPal
            </span>
          </label>
          <div
            className="icon"
          >
            <img
              alt="Paypal"
              src="IMAGE_MOCK"
            />
          </div>
        </div>
      </fieldset>
      <div
        className="paypal"
      >
        <p>
          <span>
            In order to complete your transaction, we will transfer you over PayPal's secure servers.
          </span>
        </p>
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="cardholdername"
        >
          <span>
            Cardholder name
          </span>
        </label>
        <input
          className="form-control"
          id="cardholdername"
          name="cardholdername"
          placeholder="Enter name"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="cardnumber"
        >
          <span>
            Card Number
          </span>
        </label>
        <input
          className="form-control"
          id="cardnumber"
          name="cardnumber"
          placeholder="Enter number"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="cvc"
        >
          <span>
            MM / YY CVC
          </span>
        </label>
        <input
          className="form-control"
          id="cvc"
          name="cvc"
          placeholder="mm - yy - CVC"
          type="text"
        />
      </div>
      <div
        className="form-check"
      >
        <label
          className="check_label form-check-label"
        >
          <input
            className="form-check-input"
            type="checkbox"
          />
          <span>
            Remember this card
          </span>
        </label>
      </div>
      <div
        className="summary"
      >
        <h4>
          <span>
            Summary
          </span>
        </h4>
        <ul
          clasName="list"
        >
          <li>
            <p>
              <span>
                Original Price:
              </span>
            </p>
            <p>
              $94.99
            </p>
          </li>
          <li>
            <p>
              <span>
                Coupon Discounts:
              </span>
            </p>
            <p>
              -$74.00
            </p>
          </li>
          <li>
            <p>
              <span>
                Total:
              </span>
            </p>
            <h4>
              $20.99
            </h4>
          </li>
        </ul>
      </div>
      <div
        className="term_service"
      >
        <p>
          <span>
            By completing your purchase you agree to these
          </span>
           
          <a
            href="/"
            onClick={[Function]}
          >
            <span>
              Terms of Service.
            </span>
          </a>
        </p>
      </div>
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        <span>
          Pay
        </span>
         $20.00
      </button>
      <div
        className="secure_payment"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M376 186h-20v-40c0-55-45-100-100-100S156 91 156 146v40h-20c-22.002 0-40 17.998-40 40v200c0 22.002 17.998 40 40 40h240c22.002 0 40-17.998 40-40V226c0-22.002-17.998-40-40-40zM256 368c-22.002 0-40-17.998-40-40s17.998-40 40-40 40 17.998 40 40-17.998 40-40 40zm62.002-182H193.998v-40c0-34.004 28.003-62.002 62.002-62.002 34.004 0 62.002 27.998 62.002 62.002v40z"
          />
        </svg>
        <p>
          <span>
            Payments are secure and encrypted
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
